Source/deviceinfo/DeviceInfo.cpp

```diff
@@ -401,26 +401,26 @@ class DeviceInfoLink : public Thunder::RPC::SmartInterfaceType<Thunder::Exchange
     uint32_t Deviceinfo_serial_number(char buffer[], uint8_t* length)
     {
         ASSERT(length != nullptr);
-        uint32_t result = Core::ERROR_UNAVAILABLE;
+        uint32_t result = deviceinfo_status::DEVICEINFO_ERROR_UNAVAILABLE;
         string serialNumber;
         _lock.Lock();
         if (_modelName.IsSet() == true) {
             serialNumber = _serialNumber.Value();
-            result = Core::ERROR_NONE;
+            result = deviceinfo_status::DEVICEINFO_OK;
         }
         else {
             if (_deviceInfoInterface != nullptr) {
                 result = _deviceInfoInterface->SerialNumber(serialNumber);
-                if (result == Core::ERROR_NONE) {
+                if (result == deviceinfo_status::DEVICEINFO_OK) {
```

Comment:
SerialNumber() returns code in Core::ERROR_ domain and will silently go casted to DEVICEINFO_ERROR_ yet do not match (the don't need to match). I think it requires translation (or normalization) from one to other.

Reply:
thanks for the comment let me check

Reply:
thanks, I have handled :)